discovery_engine_core/web-api/src/storage.rs

```diff
@@ -171,6 +172,27 @@ impl UserState {
 
         Ok(())
     }
+
+    #[allow(dead_code)]
+    pub(crate) async fn interacted_documents(
+        &self,
+        user_id: &UserId,
+    ) -> Result<Vec<DocumentId>, GenericError> {
+        let mut tx = self.pool.begin().await?;
+
+        let documents = sqlx::query_as::<_, QueriedInteractedDocumentId>(
+            "SELECT doc_id
```

Comment:
if I understand correctly `SELECT DISTINCT` will get rid off duplicates and our PK is (doc_id, user_id, time_stamp) which means in theory we could have same doc_id multiple times returned, yes?

Reply:
good point, it shouldn't matter much if we feed duplicates to the filter in elastic, but it's definitely better to return unique values here, i'll change it.